### PR TITLE
test(fakeredis): add topk (fakeredis) test suite

### PR DESCRIPTION
**Pull request by glevkovich**
Add a pytest suite for the TOPK data structure to validate FakeRedis compatibility against a real Redis/Dragonfly server.

- Cover HeavyKeeper commands: RESERVE, ADD, INCRBY, QUERY, COUNT, LIST, INFO, and TYPE.

- Validate probabilistic decay math and min-heap eviction logic.

- Test edge cases including zero/negative increments, parameter boundaries, missing keys, and arity mismatches.

- Ensure binary safety and proper memory lifecycle handling via DEL, EXPIRE, PEXPIRE, and RENAME.

- Include pipelined stress tests to verify memory stability.

### CODE REVIEW

File: tests/fakeredis/test/test_stack/test_topk.py [hunk 1]
```diff
@@ -1,100 +1,273 @@
...
+    r.topk().reserve("tk_ttl", 5, 8, 7, 0.9)
+
+    # Test that setting a long TTL works and reports correctly
+    assert r.expire("tk_ttl", 3600)
+    ttl = r.ttl("tk_ttl")
+    assert 0 < ttl <= 3600
+
+    # Test actual expiration mechanics, verify the sketch is completely gone
+    assert r.pexpire("tk_ttl", 100)
+    time.sleep(0.2)
```
[the remark this answers is not in the document]
Comment (glevkovich) [reply tests/fakeredis/test/test_stack/test_topk.py]: fixed

File: tests/fakeredis/test/test_stack/test_topk.py [hunk 2]
```diff
@@ -1,100 +1,273 @@
...
+            keys.append(k)
+            parsed[k] = int(items_with_count[i + 1])
+
+    if _is_real_server(r):
+        # Assert exact counts
+        for name, count in parsed.items():
+            assert count == 10, f"{name} count decayed unexpectedly to {count}"
+
+        # Assert exact lexicographical order: apple < banana < zebra
+        assert keys[0] == "apple"
```
[the remark this answers is not in the document]
Comment (glevkovich) [reply tests/fakeredis/test/test_stack/test_topk.py:136]: dropped

File: tests/fakeredis/test/test_stack/test_topk.py [hunk 3]
```diff
@@ -1,100 +1,273 @@
...
-        "B",
-        "A",
-        "B",
-        "E",
-        "E",
-    )
+        counts = r.topk().count("tk", "alpha", "beta", "gamma")
+
+    if _is_real_server(r):
+        assert counts[0] == 100
```
[the remark this answers is not in the document]
Comment (glevkovich) [reply tests/fakeredis/test/test_stack/test_topk.py:98]: fixed

File: tests/fakeredis/test/test_stack/test_topk.py [hunk 4]
```diff
@@ -1,100 +1,273 @@
...
+    threshold return None, while overpowering existing heap items correctly
+    returns the exact evicted string.
+    """
+    r.topk().reserve("tk", 2, 50, 7, 0.9)
+
+    assert r.topk().incrby("tk", ["heavy1", "heavy2"], [1000, 500]) == [None, None]
+    assert r.topk().add("tk", "weak") == [None]
+
+    # Because we gave heavy2 exactly 500, and HeavyKeeper decay is probabilistic,
+    # heavy2 is definitively the smallest and will be dropped when the 5000 weight enters.
```
[the remark this answers is not in the document]
Comment (glevkovich) [reply tests/fakeredis/test/test_stack/test_topk.py]: already fixed.
